waiter/waiter.go

```diff
@@ -0,0 +1,77 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+const (
+	// DefaultPollInterval represents duration between checks. Each DefaultPollInterval waiter make the request
+	// to confirm the operation it waits, was successfully executed.
+	// For instance: ContainerPutWaiter waits until container will be created.
+	DefaultPollInterval = 1 * time.Second
+)
+
+var (
+	// ErrConfirmationTimeout describes situation, when we sent request, but got timeout during waiting confirmation.
+	// For instance: we send containerPut, after that we wait and do containerGet, to be sure it was created successfully.
+	// If we have context expired at this step, we return this error.
```

Comment:
... Notice that in many cases this doesn't mean the operation has completely failed (because the request for it was sent without any errors).

Reply:
Added